### PR TITLE
Refactor validation and add unit tests

### PR DESCRIPTION
**Pull request by arkn98**
This PR contains a refactor of validation. I still need to write unit-tests, but before that I need your opinion on how this looks, @dyaneshvarun.

### CODE REVIEW

File: internal/controller/user_controller.go [hunk 1]
```diff
@@ -142,41 +128,13 @@ func ActivateUser() http.HandlerFunc {
...
-					validation.RuneLength(12, 128),
-					validation.By(func(value interface{}) error {
-						s, _ := value.(string)
-						if s != r.Password {
-							return errors.New("passwords do not match")
-						}
-						return nil
-					}),
-				),
+				"id":               validation.ValidateEmpID(id),
```
Comment (dyaneshvarun) [inline internal/controller/user_controller.go:134]: Quick thought here again, Do we need validation for each and every field and keep validations to be generic enough ?! Like ValidateRequiredAndNotNull

File: internal/util/validation/validation.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+	is.Digit,
+	validation.Min(4),
+	validation.Max(4),
+}
+
+// MonthPolicy is the validation policy for a month value. It must only be between 1-12.
+var MonthPolicy = []validation.Rule{
+	validation.Required,
+	is.Digit,
+	validation.Min(1),
```
Comment (dyaneshvarun) [inline internal/util/validation/validation.go:221]: Is it comparing the numbers only ?!

Comment (arkn98) [reply internal/util/validation/validation.go:221]: Yes, we have the `is.Digit` rule first. So, if it fails there, further validation is stopped.

File: internal/util/validation/validation.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+	return validation.Validate(value, rules...)
+}
+
+// ValidateResourceName validates resource names.
+func ValidateResourceName(value string) error {
+	return validation.Validate(value, ResourceNamePolicy...)
+}
+
+// ValidateResourceID validates resource ids.
+func ValidateResourceID(value string) error {
```
Comment (dyaneshvarun) [inline internal/util/validation/validation.go:293]: Can this be changed as ValidateNumber !?

File: internal/util/validation/validation.go [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+}
+
+// ValidateCustom is used to write custom validation rules for one-off use-cases.
+func ValidateCustom(value interface{}, rules ...validation.Rule) error {
+	return validation.Validate(value, rules...)
+}
+
+// ValidateResourceName validates resource names.
+func ValidateResourceName(value string) error {
+	return validation.Validate(value, ResourceNamePolicy...)
```
Comment (dyaneshvarun) [inline internal/util/validation/validation.go:289]: And this one as ValidateAlphaNumeric

**Pull request merged**